src/lib/ext-activitylog.js

```diff
@@ -4,25 +4,58 @@ class Model {
   constructor() {
     this.logs = [];
     this.filter = {
-      id: [],
-      viewType: [],
-      type: [],
+      id: new Set(),
+      viewType: new Set(),
+      type: new Set(),
+      keyword: '',
     };
   }
 
   addNewLogs(logs) {
     this.logs.push(...logs);
   }
 
-  addFilter({ logKey, valueEquals }) {
-    this.filter[logKey].push(valueEquals);
+  /**
+   * @param {object} updateFilter - It contains key and value to replace
+   * the filter
+   * @param {Set<string>} [updateFilter.id] - It contains the extension ids.
+   * @param {Set<string>} [updateFilter.viewType] - It contains view types that
+   * includes background, popup, sidebar, tab, devtools_page, devtools_panel.
+   * @param {undefined} [updateFilter.viewType] - when [updateFilter.type] is
+   * content_script.
+   * @param {Set<string>} [updateFilter.type] - It contains api types that
+   * includes api_call, api_event, content_script, user_script.
+   * @param {string} [updateFilter.keyword] - It contains a string for
```

Comment:
The description doesn't say more than the type. If you can't think of anything meaningful, just leave the description empty.